functions.go

```diff
@@ -328,3 +328,49 @@ func errorStack(err error) []string {
 	}
 	return result
 }
+
+// CauseEquals tests, if this or any cause of this error is equal to another error, or it's cause
+//
+// For example:
+//		ErrNotFound := errors.New("Not found")
+//		inSome := errors.Trace(ErrNotFound)
+//		veryDeep := errors.Trace(inSome)
+//		returned := errors.Trace(veryDeep)
+//		if errors.CauseEquals(ErrNotFound, returned) {
+//			fmt.Println("Can return soft error 404")
+//		} else {
+//			fmt.Println("Serious error, returning with code 500")
+//		}
+func CauseEquals(err1, err2 error) bool {
+	var maxDeep = 1000
+	if err1 == err2 { // Same or nil == nil
+		return true
+	}
+	if (err1 == nil && err2 != nil) || (err2 == nil && err1 != nil) {
```

Comment:
if err1 == nil || err2 == nil { .. } is sufficient, you've already asserted that err1 and err2 are not equal so therefore if err1 is nil, err2 is not and vice versa